client/src/model/values/instantaneousValues/vectorValues/VectorValue.hx

```diff
@@ -0,0 +1,148 @@
+package model.values.instantaneousValues.vectorValues;
+import model.values.instantaneousValues.vectorValues.VectorValueI;
+import model.values.instantaneousValues.InstantaneousValueI;
+import model.values.instantaneousValues.scalarValues.ScalarValueI;
+
+/**
+ * ...
+ * @author AdvaitTrivedi
+ */
+class VectorValue implements VectorValueI 
+{
+	var vector : Array<InstantaneousValueI> = new Array<InstantaneousValueI>();
+
+	public function new() 
+	{
+		
+	}
+	
+	
+	/* INTERFACE model.values.VectorValueI */
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function fromArray(instanteousValues: Array<InstantaneousValueI>) : VectorValueI {
+		for (value in instanteousValues) {
+			this.push(value);
+		}
+		
+		return this;
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function push(instantaneousValue:InstantaneousValueI, ?index:Int = 0):Void 
+	{
+		this.vector.insert(index, instantaneousValue);
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function removeFrom(?index:Int = 0):InstantaneousValueI 
+	{
+		if (index == 0) {
+			return this.vector.shift();
+		}
+		
+		if (index > this.length()) {
+			throw "Invalid index";
+		}
+		
+		var valuePopped = this.vector[index];
+		this.vector.remove(valuePopped);
+		return valuePopped;
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function concat(vectorValue:VectorValueI):Void 
+	{
+		for (value in vectorValue) {
+			this.push(value);
+		}
+	}
+	
+	@:allow(model.values.SignalValue)
+	@:allow(model.values.instantaneousValues.scalarValues.ScalarValueI)
+	function slice(?startIndex:Int = 0, ?endIndex:Int):VectorValueI 
+	{
+		if (endIndex > this.length()) {
+			throw "Invalid end index";
+		}
+		
+		var newVector = new VectorValue();
+		return newVector.fromArray(this.vector.slice(startIndex, endIndex));
+	}
+	
+	public function length():Int 
+	{
+		return this.vector.length;	
+	}
+	
+	public function depth():Int 
```

Comment:
I'm not sure that we'll ever need this method.  The depth of a scalar is 0. The depth of an empty vector is 1.  For a nonempty vector it is the 1 plus the maximum depth of its deepest member.